### DISCUSSION TITLE
Must servie register at the UI thread?

### DISCUSSION THREAD

**Discussion by Snailya** [reactiveui/ReactiveUI#3805] [Q&A]
Hello,
I'm writing an VSTO addin with WPF window. Im my addin, I have two custom threads. One is a STA for WPF, the other is a MTA for services. I create a new STA thread for WPF because that I don't want the UI blocks while doing operations in Office.
The problem is that in some of my services, it uses WhenAnyValue(). When register the service in the MTA state thread, the initialize of RxApp will try to set the MainThreadScheduler to the dispatcher for the MTA thread which is null.
How to work around it?

**Comment by anaisbetts** [comment]
I would explicitly initialize ReactiveUI on the thread you want (or even better, explicitly set `RxApp.MainThreadScheduler`)

**Reply by Snailya** [reply]
How to explicitly initialize ReactiveUI?

I found it's really strange. If I move the project out of the solution, it works well, but if it is inside, it triggers the error.

**Comment by glennawatson** [comment]
Are you targetting the correct version of windows following https://www.reactiveui.net/docs/getting-started/installation/windows-presentation-foundation.html with the TargetFramework?
